account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+                )
+                return werkzeug.utils.redirect(
+                    '/web#id={}&view_type=form&model=mtd.hello_world&'.format(record_id) +
+                    'menu_id={}&action={}'.format(menu_id.id, action.id)
+                )
+            return True
+            
+        elif (req.status_code == 401 and
+              self.which_button_type_clicked == "user" and
+              response_token['message'] == "Invalid Authentication information provided"):
+            self._logger.info(
+                "_json_command - code 401 found, user button clicked,  " +
+                "and message was:- {} ".format(response_token['message'])
+            )
+            return self.refresh_user_authorisation(token_record)
+            
+        else:
+            construct_text = (
+                "Date {}     Time {} \n".format(datetime.now().date(), datetime.now().time()) +
+                "Sorry. The connection failed ! \n" +
+                "Please check the log below for details. \n\n" +
+                "Connection Status Details: \n" +
+                "Request Sent: \n{} \n\n".format(hmrc_connection_url) +
+                "Error Code:\n{} \n\n".format(req.status_code) +
+                "Response Received: \n{}".format(response_token['message'])
+            )
+            self._logger.info("_json_command - other error found:- {} ".format(construct_text))
+            self.response_from_hmrc = construct_text
+            if record_id:
+                action = self.env.ref('account_mtd.action_mtd_hello_world')
+                menu_id = self.env.ref('account_mtd.submenu_mtd_hello_world')
+                return werkzeug.utils.redirect(
+                    '/web#id={}&view_type=form&model=mtd.hello_world'.format(record_id) +
+                    '&menu_id={}&action={}'.format(menu_id.id, action.id)
+                )
+            return True
+    
+    @api.multi
+    def get_user_authorisation(self):
+        # Update the information in the api tracker table
+        self._logger.info("(Step 1) Get authorisation")
+        tracker_api = self.env['mtd.api_request_tracker']
+        tracker_api.create({
+            'user_id': self._uid,
+            'api_id': self.api_name.id,
+            'api_name': self.api_name.name,
+            'endpoint_id': self.id,
+            'request_sent': True,
+            })
+
+        redirect_uri = "{}/auth-redirect".format(self.hmrc_configuration.redirect_url)
+        authorisation_url = "https://test-api.service.hmrc.gov.uk/oauth/authorize?"
+        self._logger.info("(Step 1) Get authorisation - authorisation URI used:- {}".format(authorisation_url))
+        state = ""
+        # State is optional
+        if self.hmrc_configuration.state:
+            state = "&state={}".format(self.hmrc_configuration.state)
+        #scope needs to be percent encoded
+        import pdb; pdb.set_trace()
+        scope = urllib.parse.quote_plus(self.scope)
+
+        authorisation_url += (
```

Comment:
I suggest declaring `authorization_url` in its entirety here, rather than declaring part of it above and appending to it here.

If you really need to log the prefix first, rename the first one to `authorization_url_prefix`, and then declare `authorization_url` here as:

```
authorization_url = (
    authorization_url
   + "next bit"
   + "next bit"
)
```

At the moment the variable changes its meaning part way through, which is one more thing to have to think about when trying to make sense of the function as an outsider.